src/main/java/com/minecolonies/blockout/loader/json/JsonLoader.java

```diff
@@ -0,0 +1,159 @@
+import net.minecraftforge.fml.relauncher.Side;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+
+public class JsonLoader implements ILoader
+{
+    @Override
+    public boolean accepts(@NotNull final ResourceLocation location)
+    {
+        return location.getResourcePath().toLowerCase().endsWith(".xml");
+    }
+
+    @Override
+    public boolean accepts(@NotNull final IPaneParams paneParams)
+    {
+        return paneParams instanceof JsonPaneParams;
+    }
+
+    @Override
+    public Pane createFromPaneParams(
+      @NotNull final IPaneParams params, @NotNull final View parent)
+    {
+        if ("layout".equalsIgnoreCase(params.getType()))
+        {
+            final String resource = params.getStringAttribute("source", null);
+            if (resource != null)
+            {
+                createFromFile(new ResourceLocation(resource), parent);
+            }
+
+            return null;
+        }
+
+        params.setParentView(parent);
+        final Pane pane = createFromPaneParams(params);
+
+        if (pane != null)
+        {
+            pane.putInside(parent);
+            pane.parseChildren(params);
+        }
+
+        return pane;
+    }
+
+    @Override
+    public void createFromFile(final ResourceLocation resource, final View parent)
+    {
+        createFromJson(createInputStream(resource), parent);
+    }
+
+    private Pane createFromPaneParams(final IPaneParams params)
+    {
+        //  Parse Attributes first, to full construct
+        final String paneType = params.getType();
+        final String style = params.getStringAttribute("style", null);
+
+        Constructor<? extends Pane> constructor = ComponentConstructionController.getConstructorForTypeAndStyle(paneType, style);
+        if (constructor == null && style != null)
+        {
+            constructor = ComponentConstructionController.getConstructorForTypeAndStyle(paneType, null);
+        }
+
+        if (constructor != null)
+        {
+            try
+            {
+                return constructor.newInstance(params);
+            }
+            catch (InstantiationException | IllegalAccessException | InvocationTargetException exc)
+            {
+                Log.getLogger().error(
+                  String.format("Exception when parsing Json for pane type %s", paneType),
+                  exc);
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Parse XML from an InputSource into contents for a View.
+     *
+     * @param input  xml file.
+     * @param parent parent view.
+     */
+    private void createFromJson(final InputStream input, final View parent)
+    {
+        final JsonParser parser = new JsonParser();
+
+        createFromJson(parser.parse(new InputStreamReader(input)).getAsJsonObject(), parent);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this use of constructor "InputStreamReader(InputStream)" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1943)